Cython/Compiler/FusedNode.py

```diff
@@ -458,6 +483,12 @@ def _buffer_declarations(self, pyx_code, decl_code, all_buffer_types):
                 except (ImportError, AttributeError, TypeError):
                     ndarray = None
             """)
+        pyx_code.func_defs.put_chunk(
+            u"""
+                def __is_little_endian():
+                    cdef int endian_detector = 1
+                    return (<char*>&endian_detector)[0] != 0
```

Comment:
I actually based this code on code that already exists in cython :) I'll take a look at what this stack overflow thread propose!